clients/bigquery/storagewrite.go

```diff
@@ -0,0 +1,200 @@
+package bigquery
+
+import (
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery/storage/apiv1/storagepb"
+	"cloud.google.com/go/bigquery/storage/managedwriter/adapt"
+	"github.com/artie-labs/transfer/lib/array"
+	"github.com/artie-labs/transfer/lib/typing"
+	"github.com/artie-labs/transfer/lib/typing/columns"
+	"github.com/artie-labs/transfer/lib/typing/decimal"
+	"github.com/artie-labs/transfer/lib/typing/ext"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/dynamicpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func columnToTableFieldSchema(column columns.Column) (*storagepb.TableFieldSchema, error) {
+	var fieldType storagepb.TableFieldSchema_Type
+	mode := storagepb.TableFieldSchema_NULLABLE
+
+	switch column.KindDetails.Kind {
+	case typing.Boolean.Kind:
+		fieldType = storagepb.TableFieldSchema_BOOL
+	case typing.Integer.Kind:
+		fieldType = storagepb.TableFieldSchema_INT64
+	case typing.Float.Kind:
+		fieldType = storagepb.TableFieldSchema_DOUBLE
+	case typing.EDecimal.Kind:
+		fieldType = storagepb.TableFieldSchema_STRING
+	case typing.String.Kind:
+		fieldType = storagepb.TableFieldSchema_STRING
+	case typing.ETime.Kind:
+		switch column.KindDetails.ExtendedTimeDetails.Type {
+		case ext.TimeKindType:
+			fieldType = storagepb.TableFieldSchema_TIME
+		case ext.DateKindType:
+			fieldType = storagepb.TableFieldSchema_DATE
+		case ext.DateTimeKindType:
+			fieldType = storagepb.TableFieldSchema_TIMESTAMP
+		default:
+			return nil, fmt.Errorf("unsupported extended time details type: %q", column.KindDetails.ExtendedTimeDetails.Type)
+		}
+	case typing.Struct.Kind:
+		fieldType = storagepb.TableFieldSchema_STRING
```

Comment:
Struct is also a string?

Reply:
Yeah we get errors if we try to use `storagepb.TableFieldSchema_JSON` here.